### DISCUSSION TITLE
View Podman Errors on Quadlet Failure?

### DISCUSSION THREAD

**Discussion by lethedata** [containers/podman#24631] [Q&A]
Is there a way to see podman specific errors when a quadlet fails to start?

When I look at `systemctl status <generated_quadlet>.service` and `journalctl -xeu <generated_quadlet>.service` after a failure all I get is generic exit-codes due to repeated too quickly but when I manually run the `podman run` command I can see the exact podman specific error. This makes troubleshooting a bit of a pain as I have to rewrite the podman command from the quadlet file manually.

**Comment by Luap99** [comment]
`journalctl -xeu <generated_quadlet>.service` should definitely already contain the same error output as it just captures stdout/err

**Reply by lethedata** [reply]
I only see what I posted above and `An ExecStart= process belonging to unit...` Failing containerized apps / running containers do show up in journalctl. It's more of malformed or unexpected podman errors.

The specific podman error I'm talking about are things related to uid/gid mapping (ie not enough range available for container to start)

**Reply by Luap99** [reply]
You need to scroll up most likely, or simply remove -x as it is pretty useless
```
Nov 21 19:42:55 hostname systemd[2998]: Starting test.service...
Nov 21 19:42:55 hostname test[77820]: Error: unsupported pull policy "abc"
Nov 21 19:42:55 hostname systemd[2998]: test.service: Main process exited, code=exited, status=125/n/a
Nov 21 19:42:56 hostname systemd[2998]: test.service: Failed with result 'exit-code'.
Nov 21 19:42:56 hostname systemd[2998]: Failed to start test.service.
```
The real podman error is clearly visible there

From an example unit with an invalid Pull= policy
```
[Container]
Image=quay.io/libpod/testimage:20240123
Exec=true
Pull=abc
```
